### PR TITLE
Android: remote device addition

### PR DESCRIPTION
**Pull request by Yashvardhan-A**
For network discovery in android we use the android library and once a server is discovered, we need to inform the remote driver about it. This function is able to do that, it registers the server in the remote driver and subsequently adds the devices to the platform.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -322,6 +322,8 @@ option(ENABLE_REMOTE_DISCOVERY_AVAHI "Enable remote disocvery in remote device f
 
 option(ENABLE_REMOTE_DISCOVERY_DHT "Enable remote disocvery in remote device for linux using DHT" OFF)
 
+option(ENABLE_REMOTE_DISCOVERY_ANDROID "Enable remote disocvery support in remote device for Android" OFF)
```
Comment (pjaaskel) [inline CMakeLists.txt:325]:
```suggestion
option(ENABLE_REMOTE_DISCOVERY_ANDROID "Enable remote discovery support in remote device for Android" OFF)
```

See also above option with the same typo.

File: lib/CL/devices/remote/network_discovery.c [hunk 2]
```diff
@@ -657,6 +658,26 @@ init_dht_discovery ()
 
 /*****************************************************************************/
 
+#if defined(ENABLE_REMOTE_DISCOVERY_ANDROID)
+
+/**
+ * Helper function used by android to add servers through the remote driver.
```
Comment (pjaaskel) [inline lib/CL/devices/remote/network_discovery.c:664]: Please run the formatter, the style seems wrong. Also separate the brief comment sentence with an empty line.

File: include/pocl_remote.h [hunk 3]
```diff
@@ -32,4 +36,28 @@
...
+#endif
+
+  /**
+   * Helper function used to dynamically add a remote server and its devices.
+   * It is currently only used by android where the network discovery is
+   * performed using android network libraries and the discovered servers are
+   * added using this function. The function is implemented in the remote
+   * driver in network_discovery.c.
+   */
+  void pocl_remote_discovery_add_server (const char *id,
```
Comment (RABijl) [inline include/pocl_remote.h:52]: is this api intended for thirdparties to use as well? a description of the parameters would be useful.

File: CMakeLists.txt (hunk 1 above)
Comment (RABijl) [reply CMakeLists.txt]: I can see this option being useful also outside of android

File: include/pocl_remote.h (hunk 3 above)
Comment (Yashvardhan-A) [reply include/pocl_remote.h:52]: I'll add the description.

Could you elaborate on the 3rd parties?

Comment (RABijl) [reply include/pocl_remote.h:52]: are only pocl developers going to use this method, or also other people?

Comment (Yashvardhan-A) [reply include/pocl_remote.h:52]: For now only pocl developers, for others to use it, it would have to be an extension.

**Review by pjaaskel — APPROVED** [review]
@RABijl can you do that small cleanup in your next PR? Otherwise this is fine to pull.

**Pull request merged**